Terminal.Gui/Application/Application.Run.cs

```diff
@@ -186,7 +186,14 @@ public static RunState Begin (Toplevel toplevel)
 
         toplevel.LayoutSubviews ();
         toplevel.PositionToplevels ();
-        toplevel.FocusFirst (null);
+
+        // TODO: Should this use FindDeepestFocusableView instead?
```

Comment:
Answering that question:

I think we should just define a `FindDeepestView` method that takes a delegate that is responsible for whatever the deepest view it's supposed to find is all about.

The basic function of finding the deepest view that is, has, or does X is the same, so duplicating that is not very DRY and is a point for bugs/regressions to creep in, as multiple disparate implementations of it diverge due to other changes.

That predicate delegate (or delegates, if appropriate) would only be responsible for filling in the blanks in the recursion for the base case and such.

Or, better yet....

If we implement IEnumerable on View, we can skip that work and use LINQ methods to do it that smarter people have already figured out how to write. 🙂

Reply:
Oh yeah totally not for this PR. I didn't think that question was scoped to this PR anyway. :)